### DISCUSSION TITLE
Packet size is incorrect

### DISCUSSION THREAD

**Discussion by xMHW** [ANLAB-KAIST/KENSv3#204] [Project Q&A 2. Transport Layer (TCP)] [answered]
![스크린샷 2024-04-25 오후 4 30 25](https://github.com/ANLAB-KAIST/KENSv3/assets/25361985/e5f9f778-0cfd-4b82-82a2-bef3a87a3297)
This is part of pcap file for running TestTransfer_Connect_Recv_Symmetric test. It says that the data length is 512 bytes so sums up to total 566 bytes (512 + 54 header bytes).
However, If I print the packet's size using the given getSize() function I get a number other than 566 bytes.
```
  // In the packetArrived function
  std::cout << "packet Size: " << packet.getSize() << std::endl;
  ```
However the log says
```
Time[1000000888]	System call [8] at [E::Host] has finished with return value [0].
Time[1000000888]	System call[syscall_no:3, unique_id: 9] has raised from app[pid:0] at [E::Host]
Time[1000000888]	[APP] write() called! sending SEQ [4294964964], size [512]
Time[1000000888]	Wire [E::Wire] received a packet [size:566] from module [E::Host]
Time[1000000888]	Wire [E::Wire] send a packet [size:566] to module [E::Switch] with transmission delay [4528], propagation delay [1000000]
Time[1001005416]	Output queue length for port[E::Wire] increased to [1]
Time[1001005416]	Output queue length for port[E::Wire] decreased to [0]
Time[1001009944]	Wire [E::Wire] received a packet [size:566] from module [E::Switch]
Time[1001009944]	Wire [E::Wire] send a packet [size:566] to module [E::Host] with transmission delay [4528], propagation delay [1000000]
Time[1002014472]	Host [E::Host] get a packet [size:566] from module [E::Wire]
packet Size: 236
```
  I cannot figure out why such discrepancy happens. Is it some bug of KENS? Or Am I missing something?
  
  For Handshaking packet,
  ```
  Time[6002592]	Host [E::Host] get a packet [size:54] from module [E::Wire]
packet Size: 54
```
It results in correct packet Size

**Comment by Pusnow** [comment]
The returned value of `packet.getSize()` for data packets should be 566.
What values do you get from the function calls?

**Reply by xMHW** [reply]
packet Size: 236 was the result that I've got

**Reply by Pusnow** [reply]
Would you send your source code to TAs? cs341.pj@kaist.ac.kr

**Reply by xMHW** [reply]
Oh well, I figured out that it was a matter of formatting of print.
```
  std::printf("packet Size: %zu\n", packet.getSize());
  ```
  I changed to this printing code and now it says the correct value(566)
  ```
  Time[1002014472]	Host [E::Host] get a packet [size:566] from module [E::Wire]
packet Size: 566
```

**Reply by Pusnow** [reply, marked as answer]
Right. Decimal 566 is hexadecimal 0x236.
